### PR TITLE
Add logging when optimizing delete to metadata-only-delete

### PR DESCRIPTION
**Pull request by szehon-ho**
It's not easy for users to know whether they are writing deletes that can be performed with just metadata, so adding this log line.

Example output from a unit test:
```
org.apache.spark.sql.execution.datasources.v2.OptimizeMetadataOnlyDeleteFromTable - Optimizing delete expression: IsNull(dep) as metadata delete
```

### CODE REVIEW

**Comment by szehon-ho** [comment]
Note: on my first version I tried to redact the log message for any sensitive filters, but not sure if its worth the complexity as there's other places in core Iceberg where filters are logged.  But let me know if that's needed and I can look at it.

File: spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/OptimizeMetadataOnlyDeleteFromTable.scala [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 package org.apache.spark.sql.execution.datasources.v2
 
+import org.apache.spark.internal.Logging
```
Comment (rdblue) [inline spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/OptimizeMetadataOnlyDeleteFromTable.scala:22]: We should not use Spark's internal logging. Can you convert this to use SLF4J instead?

**Comment by rdblue** [comment]
I think it's fine not to redact the filters. Thanks for working on this, @szehon-ho!

File: spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/OptimizeMetadataOnlyDeleteFromTable.scala (hunk 1 above)
Comment (szehon-ho) [reply spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/OptimizeMetadataOnlyDeleteFromTable.scala]: Done.  

Interestingly, turns out this object was actually extending 'logging' trait transitively and so already had 'log' variable in scope.  Named the SLF4J logger used here as 'logger'.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @szehon-ho!

**Pull request merged**